UI/src/view/AssetFormFactory.js

```diff
@@ -952,29 +894,121 @@
       return sideCode ? _assetTypeConfiguration.className + '-' + sideCode : _assetTypeConfiguration.className;
     }
 
-    function validateAdministrativeClass(selectedLinearAsset, editConstrains){
-      editConstrains = editConstrains || function() { return false; };
+    function checkEditConstrains(selectedAsset){
+      var editConstrains = _assetTypeConfiguration.editConstrains || function() { return false; };
 
-      var selectedAssets = _.filter(selectedLinearAsset.get(), function (selected) {
-        return editConstrains(selected);
+      var selectedAssets = _.filter(selectedAsset.get(), function (asset) {
+        return editConstrains(asset);
       });
       return !_.isEmpty(selectedAssets);
     }
 
-    var renderLinktoWorkList = function renderLinktoWorkList(layerName) {
-      var textName;
-      switch(layerName) {
-        case "maintenanceRoad":
-          textName = "Tarkistamattomien huoltoteiden lista";
-          break;
-        default:
-          textName = "Vanhentuneiden kohteiden lista";
+    me.isSplitOrSeparatedAllowed = function(){
+      var numberFields = function(sideCode) {
```

Comment:
This function don't count the number of fields it should be called number of property values.